src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/ParserUtil.java

```diff
@@ -49,6 +49,23 @@ public static Title parseTitle(String title) throws ParseException {
         return new Title(trimmedTitle);
     }
 
+    /**
+     * Parses a {@code String title} into a {@code Title}.
+     * Leading and trailing whitespaces will be trimmed.
+     * Any extra whitespaces between words will be removed.
+     *
+     * @throws ParseException if the given {@code title} is invalid.
+     */
+    public static Title parseTitleWithAdditionalWhitespaces(String title) throws ParseException {
+        requireNonNull(title);
+        String removedExtraWhiteSpacesTitle = title.replaceAll("\\s{2,}", " ");
+        String trimmedTitle = removedExtraWhiteSpacesTitle.trim();
+        if (!Title.isValidTitle(trimmedTitle)) {
+            throw new ParseException(Title.MESSAGE_CONSTRAINTS);
+        }
+        return new Title(trimmedTitle);
+    }
```

Comment:
Perhaps it would be better for this method to simple do the `replaceAll`, then pass `removedExtraWhiteSpacesTitle` into `parseTitle`? That way, we get less repetition of code.

Also, I believe `whitespace` is a single word.